src/containers/ListingPage/ListingPage.jsx

```diff
@@ -18,23 +18,48 @@ import ViewBar from './components/ViewBar';
 
 class ListingPage extends Component {
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      viewType: 'grid',
+    };
+    this.onViewChangeToList = this.onViewChangeToList.bind(this);
+    this.onViewChangeToGrid = this.onViewChangeToGrid.bind(this);
+    this.viewType = this.viewType.bind(this);
+  }
+
   componentWillMount() {
-    const { fetchListing, params: { listingId } } = this.props;
-    fetchListing(listingId);
+    const { fetchListingByFilter, params: { listingId } } = this.props;
+    fetchListingByFilter(listingId);
+  }
+
+  onViewChangeToList() {
```

Comment:
Mulig jeg gjør det feil, men den går helt bananas hvis jeg prøver å endre staten der med:

```
warning.js?8a56:36 Warning: setState(...): Cannot update during an existing state transition (such as within `render` or another component's constructor). Render methods should be a pure function of props and state; constructor side-effects are an anti-pattern, but can be moved to `componentWillMount`.
```

Husker nå at det var derfor jeg endte opp med to metoder, fordi det laga den ikke noe bråk om.